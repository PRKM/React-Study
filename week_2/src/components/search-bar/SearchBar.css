input,
input:focus {
    width: 100%;
    border: 3px solid rgb(0, 101, 243);
    background-color: rgb(86, 153, 247);
    color: #fff;
    padding: 12px 20px;
    font-size: 20px;
    outline: none;
}