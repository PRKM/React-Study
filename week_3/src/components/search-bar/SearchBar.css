input,
input:focus {
    width: 250px;
    border: 3px solid rgb(0, 101, 243);
    background-color: rgb(86, 153, 247);
    color: #fff;
    padding: 10px;
    font-size: 14px;
    outline: none;
}