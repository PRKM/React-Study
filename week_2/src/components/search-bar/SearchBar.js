import React from 'react';
import PropTypes from 'prop-types';

import './SearchBar.css';

export default class SearchBar extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div>
                <input type="text" value={this.props.text} onChange={(e) => {this.props.onChange(e.target.value)}}/>
            </div>
        )
    }
}

SearchBar.propTypes = {
    text: PropTypes.string,
    onChange: PropTypes.func
};