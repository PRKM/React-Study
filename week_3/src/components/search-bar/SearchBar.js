import React from 'react';
import PropTypes from 'prop-types';

import './SearchBar.css';

const SearchBar = function(props) {
    let { column, value, onChange } = props;
    return (
        <div>
            <input type="search" value={value} placeholder={column} onChange={(e) => {onChange(column, e.target.value)}}/>
        </div>
    );
}

SearchBar.propTypes = {
    column: PropTypes.string,
    value: PropTypes.string,
    onChange: PropTypes.func
};

export default SearchBar;