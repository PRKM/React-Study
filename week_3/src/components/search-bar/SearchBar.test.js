import React from 'react';
import {shallow} from 'enzyme';

import SearchBar from './SearchBar';

import 'isomorphic-fetch';

describe('<SearchBar/>', () => {
    let onChange;

    beforeAll(() => {
        onChange = (column, value) => console.log(`${column}: ${value}`);
    })

    it('Init render', () => {
        let component = shallow(<SearchBar column="country" value="" onChange={onChange}/>);

        expect(component).toMatchSnapshot();
    });

    it('Init state', () => {
        let component = shallow(<SearchBar column="country" value="" onChange={onChange}/>);

        // expect(component.prop('column')).toEqual('country');
        expect(component.prop('value')).toEqual('');
        // expect(component.prop('onChange')).toBeDefined();
    })
})