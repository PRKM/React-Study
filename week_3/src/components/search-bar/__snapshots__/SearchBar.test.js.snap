// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar/> Init render 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="country"
    type="search"
    value=""
  />
</div>
`;
